soh/soh/Enhancements/controls/GameControlEditor.h

```diff
@@ -1,7 +1,18 @@
 #pragma once
 
+#include <libultraship/libultraship.h>
+
 namespace GameControlEditor {
-    static int CurrentPort = 0;
-    static int BtnReading = -1;
-    void Init();
-}
+class GameControlEditorWindow : public LUS::GuiWindow {
+    public:
+        using LUS::GuiWindow::GuiWindow;
+
+        void InitElement() override;
+        void DrawElement() override;
+        void UpdateElement() override {};
```

Comment:
I see a mix here and in other places of `override;` and `override {};`, is there any difference?

Reply:
we need to define all 3 of those methods since they're pure virtual on the base class

https://github.com/Kenix3/libultraship/blob/6d29ff2dbc4507b59771d5af84cb58eaacd2f9a7/src/window/gui/GuiElement.h#L24-L27

we have definitions for `InitElement` and `DrawElement` in `GameControlEditor.cpp`, but we don't use `UpdateElement` so we just define it to be an empty method in the header here